### PR TITLE
Implement pcl support

### PR DESCRIPTION
**Pull request by DangMH**
- Organizes solution into PCL support directory structure
- Moves Bindings into HockeyApp.Android and HockeyApp.iOS namespaces
- Shared PCL will be in HockeyApp namespace and exposes the following
  - MetricsManager  _*Requires MetricsManager to be initialized in platform specific project_
    - .Disabled
    - .TrackEvent(string eventName)
  - CrashManager
    - . DidCrashInLastSession  _*Requires CrashManager to be initialized in platform specific project_
- Updates documentation
- Adds Xamarin.Forms sample project

### CODE REVIEW

**Comment by DangMH** [comment]
@chrwend it looks like the CI is failing because of a missing provisioning profile for building the sample projects.

**Comment by chrwend** [comment]
@DangMH: Looks good to me. Did testing on Android as well as iOS. Please make sure to add some notes on how to use the SDK within PCLs.

**Comment by DangMH** [comment]
@chrwend I've added documentation and a sample project as suggested.  If you could try the sample out for me that'd be great.  Otherwise, looks like we should be good to go.

File: README.md [hunk 1]
```diff
@@ -397,6 +409,14 @@ HockeyLog.LogLevel = 5; // Warn, show warnings and errors
 HockeyLog.LogLevel = 6; // Error, show only errors – the default log level
 ```
 
+<a id="forms-integrate-sdk"></a>
+### 3.4 Xamarin.Forms Project Integrate HockeySDK
+When adding HockeySDK-Xamarin to a Xamarin.Forms PCL Solution, add the nuget to the PCL, iOS, and Android project (Windows phone is not currently supported).
+
+Initialization must be done in the individual iOS and Android projects as shown in [Integrate HockeySDK](#integrate-sdk).  The PCL project will have access to a subset of shared non-configuration/initialization features.
+
+Please refer to the Xamarin.Forms sample in /samples/HockeyAppSampleForms.sln
```
Comment (chrwend) [inline README.md]: missing dot

File: README.md [hunk 2]
```diff
@@ -397,6 +409,14 @@ HockeyLog.LogLevel = 5; // Warn, show warnings and errors
 HockeyLog.LogLevel = 6; // Error, show only errors – the default log level
 ```
 
+<a id="forms-integrate-sdk"></a>
+### 3.4 Xamarin.Forms Project Integrate HockeySDK
+When adding HockeySDK-Xamarin to a Xamarin.Forms PCL Solution, add the nuget to the PCL, iOS, and Android project (Windows phone is not currently supported).
+
+Initialization must be done in the individual iOS and Android projects as shown in [Integrate HockeySDK](#integrate-sdk).  The PCL project will have access to a subset of shared non-configuration/initialization features.
```
Comment (chrwend) [inline README.md]: double space after `(#integrate-sdk).`

**Comment by chrwend** [comment]
Seems like the ressource bundle is not working properly in the new sample project. For example it only shows the keys but not the actual translations in the send crash confirmation alert (iOS)

**Comment by chrwend** [comment]
Had a second look at the sample project, this time translations showed up correctly. Feel free to merge and release whenever you are ready

**Pull request merged**